pool/web/handler/helpers_partial_update.ml

```diff
@@ -0,0 +1,197 @@
+      in
+      let htmx_element () =
+        let hx_post =
+          Htmx.(
+            if is_admin
+            then admin_profile_hx_post (Contact.id contact)
+            else contact_profile_hx_post)
+          |> path_with_lang
+          |> Sihl.Web.externalize_path
+        in
+        let open Pool_common.Message in
+        match partial_update with
+        | Ok partial_update ->
+          partial_update
+          (* CustomFields: It updates on error?? *)
+          |> Contact.PartialUpdate.increment_version
+          |> fun m ->
+          Htmx.partial_update_to_htmx
+            language
+            tenant_languages
+            m
+            ~hx_post
+            ~success:true
+            ()
+          |> html_response
+        | Error error ->
+          let create_htmx ?htmx_attributes ?(field = field) value =
+            Htmx.create
+              (Htmx.create_entity ?htmx_attributes version field value)
+              language
+              ~hx_post
+              ~error
+              ()
+            |> html_response
+          in
+          (match[@warning "-4"] field with
+           | Field.Firstname ->
+             Htmx.Text (value |> CCOption.pure) |> create_htmx
+           | Field.Lastname -> Htmx.Text (value |> CCOption.pure) |> create_htmx
+           | Field.Paused -> Htmx.Checkbox false |> create_htmx
+           | Field.Language ->
+             Htmx.Select
+               Htmx.
+                 { show = Pool_common.Language.show
+                 ; options = tenant_languages
+                 ; selected =
+                     value |> Pool_common.Language.create |> CCResult.to_opt
+                 }
+             |> create_htmx
+           | _ ->
+             let open Custom_field in
+             let%lwt field =
+               let open Utils.Lwt_result.Infix in
+               field_id
+               |> CCOption.to_result InvalidHtmxRequest
+               |> Lwt_result.lift
+               >>= find_by_contact tenant_db (Contact.id contact)
+             in
+             (match field with
+              | Error error ->
+                HttpUtils.(
+                  htmx_redirect
+                    back_path
+                    ?query_language
+                    ~actions:[ Message.set ~error:[ error ] ]
+                    ())
+              | Ok field ->
+                let value =
+                  match field with
+                  | Public.Number _ -> value |> CCInt.of_string |> Htmx.number
+                  | Public.Text _ -> value |> CCOption.pure |> Htmx.text
+                in
+                Htmx.custom_field_to_htmx
+                  ~value
+                  language
+                  field
+                  ~hx_post
+                  ~error
+                  ()
+                |> html_response))
+      in
+      let%lwt () =
+        match events with
+        | Error _ -> Lwt.return_unit
```

Comment:
Yes, the update handler cannot return an error. It is only a result type as all cqrs command handlers return errors. The validation is done before as it is a Lwt.

If an error occurs, it will be shown inside the Html element returned by htmx.